code/modules/mob/living/simple_animal/friendly/nian_caterpillar.dm

```diff
@@ -0,0 +1,157 @@
+	voice_name = "nian caterpillar"
+	speak_emote = list("flutters", "chitters", "chatters")
+	emote_hear = list("flutters", "chitters", "chatters")
+	emote_see = list("flutters", "chitters", "chatters")
+
+	// Special verbs for when someone interacts with a caterpillar
+	response_help  = "pets"
+	response_disarm = "shoos"
+	response_harm   = "kicks"
+
+	// Xenobiology and cargo are the only ways to get the caterpillar.
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+	melee_damage_lower = 5
+	melee_damage_upper = 8
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+
+	holder_type = /obj/item/holder/nian_caterpillar
+	can_collar = TRUE
+
+	/// Evolution action.
+	var/datum/action/innate/nian_caterpillar_emerge/evolve_action = new()
+	/// The amount of nutrition the nian caterpillar needs to evolve.
+	var/nutrition_need = 500
+
+/mob/living/simple_animal/nian_caterpillar/Initialize()
+	. = ..()
+	real_name = name
+	add_language("Tkachi")
+	evolve_action.Grant(src)
+
+/mob/living/simple_animal/nian_caterpillar/proc/evolve(obj/structure/moth/cocoon/C, datum/mind/M)
+	if(stat != CONSCIOUS)
+		return FALSE
+
+	// A changeling caterpillar shouldn't be restricted from evolving.
+	// A caterpillar needs to consume food-- similar to a dioan nymph --to evolve.
+	if((nutrition < nutrition_need) && !IS_CHANGELING(M))
+		to_chat(src, "<span class='warning'>You need to binge on food in order to have the energy to evolve...</span>")
+		return
+
+	// Worme is the lesser form of nian. The caterpillar evolves into this lesser form.
+	var/mob/living/carbon/human/nian_worme/adult = new(get_turf(loc))
+
+	if(istype(loc, /obj/item/holder/nian_caterpillar))
+		var/turf/cocoon_turf = get_turf(loc)
+		qdel(loc)
+		forceMove(cocoon_turf)
+
+	for(var/datum/language/L in languages)
+		adult.add_language(L.name)
+	adult.regenerate_icons()
+	adult.name = src.name
+	adult.real_name = adult.name
+
+	// Mind transfer to new worme.
+	M.transfer_to(adult)
+
+	// Worme is placed into cacoon.
+	adult.forceMove(C)
+	C.preparing_to_emerge = TRUE
+	adult.apply_status_effect(STATUS_EFFECT_COCOONED)
+	adult.KnockOut() // Zzzz
+	adult.create_log(MISC_LOG, "has woven a cocoon")
+
+	// For any random generated names. This is for when a new nian caterpillar is spawned.
+	// [ nian caterpillar (042) ] etc.
+	if(findtext(adult.real_name, "nian caterpillar"))
+		adult.real_name = adult.dna.species.get_random_name()
+		adult.name = adult.real_name
+	for(var/obj/item/W in contents)
+		unEquip(W)
+
+	qdel(src)
+	return TRUE
+
+/mob/living/simple_animal/nian_caterpillar/proc/consume(obj/item/food/snacks/G)
+	if(nutrition >= nutrition_need) // Prevents griefing by overeating food items without evolving.
+		return to_chat(src, "<span class='warning'>You're too full to consume this! Perhaps it's time to grow bigger...</span>")
+	visible_message("[src] ravenously consumes [G].", "You ravenously devour [G].")
+	playsound(loc, 'sound/items/eatfood.ogg', 30, TRUE, frequency = 1.5)
```

Comment:
I think frequency is in hz, are you sure this will do what you want it to?

Reply:
Yes, when played in game is sounded fine. It's the exact same as the diona nymph eating sound.